src/tests/unit/tests/injected/components/details-dialog.test.tsx

```diff
@@ -68,8 +70,24 @@ describe('DetailsDialogTest', () => {
             helpUrl,
             snippet: 'html',
         };
-        const expectedFailedRules: DictionaryStringTo<DecoratedAxeNodeResult> = {};
-        expectedFailedRules[ruleId] = expectedNodeResult;
+        return expectedNodeResult;
+    }
+
+    function testRender(
```

Comment:
It might be the way GitHub is displaying the diff here, but I'm finding the creation of the test rules data more difficult to follow than I would expect (some of this is from the original code). There's parameter `expectedFailedRules` and local `_expectedFailedRules` (neither of which are used in an `expect`) and a `createFailedRules` helper (which doesn't generate either of the previous things)). Might it be cleaner to organize as:

- `createFailedRules` helper takes a number of rules to create (presumably 1 or 2)
- `testRender`'s last parameter is a mandatory number of rules to create
- `failedRules` is a local in `testRender` and is the result of calling `createFailedRules` with the rules count passed in

(or alternately callers to `testRender` pass as last parameter the result of calling `createFailedRules` with the desired count)